src/trace/layers/ordered.rs

```diff
@@ -38,36 +39,37 @@ impl<K: Ord+Clone, L: Trie> Trie for OrderedLayer<K, L> {
 			let child_lower = self.offs[lower];
 			let child_upper = self.offs[lower + 1];
 			OrderedCursor {
-				keys: self.keys.clone(),
-				offs: self.offs.clone(),
+				keys: owned_self.clone().map(|x| &x.keys),
+				offs: owned_self.clone().map(|x| &x.offs),
```

Comment:
Also, getting to `OwningRef` was a journey through (arguably) desirable but not-yet-implemented rust type systems features, namely higher-kinded-types or associated type constructors, which could have enabled just using lifetimes and references without most of the pain.